### DISCUSSION TITLE
What is the correct way of calculating the correct position using `project()` with the `onPaneContextMenu` ?

### DISCUSSION THREAD

**Discussion by RiadhAdrani** [xyflow/xyflow#3209] [Q&A] [answered]
What is the correct way of calculating the correct position using `project()` with the `onPaneContextMenu` ?

I am unable to find a guide or a section answering this question, I found [this](https://github.com/wbkd/react-flow/issues/1176) issue, but it there is an annoying offset, any suggestions ?

Thanks.

**Comment by moklick** [comment, marked as answer]
Hey,

you need to pass the position relative to the ReactFlow container to the `project` function. 

In your onPaneContextMenu handler you could do something like:
```js
const reactFlowBounds = document.querySelector('.react-flow').getBoundingClientRect(); // this could also be done with a ref
const position = project({
  x: event.clientX - reactFlowBounds.left,
  y: event.clientY - reactFlowBounds.top,
});
```
